nixos/modules/programs/firefox.nix

```diff
@@ -49,42 +51,106 @@ in {
         ${organisationInfo}
       '';
       type = policyFormat.type;
-      default = {};
+      default = { };
     };
 
     preferences = mkOption {
       description = mdDoc ''
-        Preferences to set from `about://config`.
+        Preferences to set from `about:config`.
 
         Some of these might be able to be configured more ergonomically
         using policies.
 
         ${organisationInfo}
       '';
       type = with types; attrsOf (oneOf [ bool int string ]);
-      default = {};
+      default = { };
+    };
+
+    preferencesStatus = mkOption {
+      description = mdDoc ''
+        The status of `firefox.preferences`.
+
+        `status` can be "default", "locked", "user" or "clear"
+        - `"default"`: Read/Write: Settings appear as default even if factory default differs.
+        - `"locked"`: Read-Only: Settings appear as default even if factory default differs.
+        - `"user"`: Read/Write: Settings appear as changed if it differs from factory default.
+        - `"clear"`: Read/Write: Value has no effect. Resets to factory defaults on each startup.
+      '';
+      type = types.enum [ "default" "locked" "user" "clear" ];
+      default = "locked";
```

Comment:
Normally we would order type, default and description but all other things are like this too, so whatever.